<!-- Top-Level launch file for bundled launching /-->
<!--Simulation launcher for the HCAUV aircraft-->

<launch>
	<!-- Available arguments -->
	<arg name="manualMode" default="true"/>
	<arg name="autoMode" default="false"/>
	<arg name="ArduPlane" default="false"/>
	<arg name="uav_name" default="skywalker_2013"/>
	<arg name="log" default="false"/>
	<!-- <arg name="groundCam" default="false"/> -->

	<!-- External optional arguments -->
	<arg name="home" default="nan"/>
	<arg name="simRate" default="nan"/>
	<arg name="deltaT" default="nan"/>

	<!-- Enable simulation clock -->
	<param name="use_sim_time" type="bool" value="true" />

	<!-- Launch the simulator /-->
	<include file="$(find last_letter)/launch/simulation.launch" >
		<arg name="uav_name" value="$(arg uav_name)"/>
		<arg name="log" value="$(arg log)"/>
		<arg name="home" value="$(arg home)"/>
		<arg name="simRate" value="$(arg simRate)"/>
		<arg name="deltaT" value="$(arg deltaT)"/>
		<arg name="ArduPlane" value="$(arg ArduPlane)"/>
	</include>

	<!-- Launch the input and control nodes -->
	<include file="$(find last_letter)/launch/controls.launch" >
		<arg name="manualMode" value="$(arg manualMode)"/>
		<arg name="autoMode" value="$(arg autoMode)"/>
		<arg name="uav_name" value="$(arg uav_name)"/>
		<arg name="ArduPlane" value="$(arg ArduPlane)"/>
	</include>

	<!-- ArduPlane SITL connection -->
	<group if="$(arg ArduPlane)" ns="$(arg uav_name)">
		<node pkg="last_letter" type="fdmUDPSend.py" name="fdmUDPSend"  output="screen"/>
		<node pkg="last_letter" type="ctrlInjector" name="injector" output="screen"/>
	</group>

	<!-- Launch the visualizers /-->
	<include file="$(find last_letter)/launch/visualization.launch" >
		<arg name="uav_name" value="$(arg uav_name)" />
		<!-- <arg name="groundCam" value="$(arg groundCam" /> -->
	</include>

</launch>